examples/face_recognition_11/face_recognition_11/evaluator.py

```diff
@@ -0,0 +1,392 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+import numpy as np
+from face_recognition_11.utils import calculate_mse_nmse
+from face_recognition_11.utils import compute_baseline_thresholds
+from face_recognition_11.utils import compute_distances
+from face_recognition_11.utils import compute_pair_metrics
+from face_recognition_11.utils import compute_threshold
+from face_recognition_11.utils import create_iou_histogram
+from face_recognition_11.utils import create_similarity_histogram
+from face_recognition_11.workflow import GroundTruth
+from face_recognition_11.workflow import Inference
+from face_recognition_11.workflow import PairSample
+from face_recognition_11.workflow import PerBBoxMetrics
+from face_recognition_11.workflow import PerKeypointMetrics
+from face_recognition_11.workflow import TestCase
+from face_recognition_11.workflow import TestCaseMetrics
+from face_recognition_11.workflow import TestSample
+from face_recognition_11.workflow import TestSampleMetrics
+from face_recognition_11.workflow import TestSuiteMetrics
+from face_recognition_11.workflow import ThresholdConfiguration
+
+from kolena.workflow import AxisConfig
+from kolena.workflow import BarPlot
+from kolena.workflow import Curve
+from kolena.workflow import CurvePlot
+from kolena.workflow import EvaluationResults
+from kolena.workflow import Plot
+from kolena.workflow import TestCases
+from kolena.workflow.metrics import f1_score
+from kolena.workflow.metrics import iou
+from kolena.workflow.metrics import precision
+from kolena.workflow.metrics import recall
+
+
+def compute_per_sample(
+    ground_truth: GroundTruth,
+    inference: Inference,
+    threshold: float,
+    configuration: ThresholdConfiguration,
+    test_sample: TestSample = None,
+) -> TestSampleMetrics:
+    # Stage 1: Detection
+    if inference.bbox is None:
+        bbox_fte = True
+
+    iou_value = (
+        iou(ground_truth.bbox, inference.bbox)
+        if (ground_truth.bbox is not None and inference.bbox is not None)
+        else None
+    )
+    bbox_fte, tp, fp, fn = False, False, False, False
+
+    if iou_value is not None:
```

Comment:
If `inference.box` is None, `iou_value` becomes None therefore this block of the code doesn't get run. In which case, we are not counting FN properly.